### ISSUE TITLE
Java build - strange behavior of javac

### ISSUE THREAD

**Issue by ttusar** [numbbo/coco#1910]
The java build using the current development branch fails on a Windows computer with the following error:

````
(py36coco) D:\Delo\Github\numbbo-coco>python do.py build-java
AML     ['code-experiments/src/coco_random.c', 'code-experiments/src/coco_suite.c', 'code-experiments/src/coco_observer.c', 'code-experiments/src/coco_archive.c', 'code-experiments/src/coco_runtime_c.c'] -> code-experiments/build/java/coco.c
EXPAND  code-experiments/build/java/coco.c.in to code-experiments/build/java/coco.c
EXPAND  code-experiments/src/coco.h to code-experiments/build/java/coco.h
WRITE   code-experiments/build/java/REVISION
WRITE   code-experiments/build/java/VERSION
RUN     javac -h . CocoJNI.java in code-experiments/build/java
RUN     x86_64-w64-mingw32-gcc -I C:\Program Files\Java\jdk-12.0.2\include -I C:\Program Files\Java\jdk-12.0.2\include\win32 -shared -o CocoJNI.dll CocoJNI.c in code-experiments/build/java
RUN     javac Problem.java in code-experiments/build/java
ERROR: return value=1
Problem.java:31: error: cannot find symbol
                        this.dimension = CocoJNI.cocoProblemGetDimension(pointer);
                                         ^
  symbol:   variable CocoJNI
  location: class Problem

[... etc ...]

Problem.java:130: error: cannot find symbol
                return (CocoJNI.cocoProblemIsFinalTargetHit(pointer) == 1);
                        ^
  symbol:   variable CocoJNI
  location: class Problem
15 errors

Traceback (most recent call last):
  File "do.py", line 1025, in <module>
    main(sys.argv[1:])
  File "do.py", line 989, in main
    elif cmd == 'build-java': build_java()
  File "do.py", line 708, in build_java
    run('code-experiments/build/java', ['javac', 'Problem.java'], verbose=_verbosity)
  File "D:\Delo\Github\numbbo-coco\code-experiments\tools\cocoutils.py", line 121, in run
    universal_newlines=True)
  File "D:\Delo\Github\numbbo-coco\code-experiments\tools\cocoutils.py", line 43, in check_output_with_print
    output = check_output(*popenargs, **kwargs)
  File "C:\Anaconda3\envs\py36coco\lib\subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "C:\Anaconda3\envs\py36coco\lib\subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['javac', 'Problem.java']' returned non-zero exit status 1.
````
I was playing with this for a bit and while the command 
````
javac code-experiments\build\java\Problem.java
````
fails with the same error, the following one works just fine and compiles all classes, including the Problem class:
````
javac code-experiments\build\java\*.java
````
Even if I do 
````
javac code-experiments\build\java\Problem.java
````
again after all the classes have already been built, this command fails.

Go figure... In any case, for me this means that the java build using `do.py` fails.

**Comment by ttusar** [comment]
On another Windows computer I encountered problems even before:

````
(py36coco) D:\Users\Tea\numbbo-coco>gcc -I "C:\Program Files\Java\jdk-10.0.1\include" -I "C:\Program Files\Java\jdk-10.0.1\include\win32" -shared -o code-experiments\build\java\CocoJNI.dll code-experiments\build\java\CocoJNI.c
In file included from C:\Program Files\Java\jdk-10.0.1\include/jni.h:45:0,
                 from code-experiments\build\java\CocoJNI.c:12:
C:\Program Files\Java\jdk-10.0.1\include\win32/jni_md.h:35:9: error: unknown type name '__int64'
 typedef __int64 jlong;
         ^~~~~~~
````

Hits for a fix: https://stackoverflow.com/questions/27891478/error-when-compiling-in-cygwin

**Comment by ttusar** [comment]
Ah, the problem on this other computer had to do with the compiler (no need to fix anything in COCO). And now that I have sorted this out, the java build works on this other Windows machine. Not on my own computer though...

**Comment by ttusar** [comment]
I can make compilation work using `javac code-experiments\build\java\*.java`. 
However, running the example experiment fails:
````
(py36coco) D:\Delo\Github\ttusar-coco>python do.py verbose run-java
AML     ['code-experiments/src/coco_random.c', 'code-experiments/src/coco_suite.c', 'code-experiments/src/coco_observer.c', 'code-experiments/src/coco_archive.c', 'code-experiments/src/coco_runtime_c.c'] -> code-experiments/build/java/coco.c
EXPAND  code-experiments/build/java/coco.c.in to code-experiments/build/java/coco.c
EXPAND  code-experiments/src/coco.h to code-experiments/build/java/coco.h
WRITE   code-experiments/build/java/REVISION
WRITE   code-experiments/build/java/VERSION
RUN     javac -h . CocoJNI.java in code-experiments/build/java

RUN     x86_64-w64-mingw32-gcc -I C:\Program Files\Java\jdk-12.0.2\include -I C:\Program Files\Java\jdk-12.0.2\include\win32 -shared -o CocoJNI.dll CocoJNI.c -lwsock32 in code-experiments/build/java

RUN     javac *.java in code-experiments/build/java

RUN     java -Djava.library.path=. ExampleExperiment in code-experiments/build/java
ERROR: return value=1
Error: Could not find or load main class ExampleExperiment
Caused by: java.lang.ClassNotFoundException: ExampleExperiment
````
The ExampleExperiment.class was created and is in the right folder

**Comment by ttusar** [comment]
Should be fixed by #1911
